### PR TITLE
upper: if tiltfile breaks and then unbreaks, clear error [ch603]

### PR DESCRIPTION
**Pull request by maiamcc**
Hello @landism, @jazzdan,

Please review the following commits I made in branch maiamcc/break-unbreak-tiltfile:

da45038a902b0600599552d4b2650defd2f7b142 (2018-10-18 11:31:44 -0400)
upper: if tiltfile breaks and then unbreaks, clear error

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/engine/upper.go [hunk 1]
```diff
@@ -612,17 +617,17 @@ func eventContainsConfigFiles(manifest model.Manifest, e manifestFilesChangedAct
 	return false
 }
 
-func getNewManifestFromTiltfile(ctx context.Context, name model.ManifestName) (model.Manifest, error) {
+func getNewManifestFromTiltfile(name model.ManifestName) (model.Manifest, *newManifestErr) {
 	t, err := tiltfile.Load(tiltfile.FileName, os.Stdout)
 	if err != nil {
-		return model.Manifest{}, err
+		return model.Manifest{}, newManifestErrf(err.Error())
```
Comment (maiamcc) [inline internal/engine/upper.go]: teeeechnically we're losing some error history here by not wrapping this... do we care?

File: internal/engine/upper.go [hunk 2]
```diff
@@ -700,3 +705,15 @@ func showError(ctx context.Context, state *store.EngineState, resourceNumber int
...
+
+type newManifestErr struct {
+	s string
+}
+
+func (e *newManifestErr) Error() string { return e.s }
+
+var _ error = &newManifestErr{}
+
+func newManifestErrf(format string, a ...interface{}) *newManifestErr {
```
Comment (jazzdan) [inline internal/engine/upper.go]: I think we could remove the assertion above by changing this line to:

```suggestion
func newManifestErrf(format string, a ...interface{}) error {
```

File: internal/engine/upper.go [hunk 3]
```diff
@@ -612,17 +617,17 @@ func eventContainsConfigFiles(manifest model.Manifest, e manifestFilesChangedAct
 	return false
 }
 
-func getNewManifestFromTiltfile(ctx context.Context, name model.ManifestName) (model.Manifest, error) {
+func getNewManifestFromTiltfile(name model.ManifestName) (model.Manifest, *newManifestErr) {
```
Comment (jazzdan) [inline internal/engine/upper.go]: Which would mean changing this to an `error` too.

Comment (maiamcc) [reply internal/engine/upper.go]: i sorta like having `newManifestError` in the signature so ppl don't ignore it and add a clause later that returns some other `error` -- but also i don't have strong feelings.

**Pull request merged**